presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayAllocator.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.block;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+import static java.lang.String.format;
+
+@NotThreadSafe
+public class ArrayAllocator
+{
+    private static final int DEFAULT_MAX_OUTSTANDING = 100;
+
+    private final Deque<byte[]> byteArrays = new ArrayDeque<>();
+    private final Deque<short[]> shortArrays = new ArrayDeque<>();
+    private final Deque<int[]> intArrays = new ArrayDeque<>();
+    private final Deque<long[]> longArrays = new ArrayDeque<>();
+    private final Deque<boolean[]> booleanArrays = new ArrayDeque<>();
+
+    private int outstandingArrays;
+    private final int maxOutstandingArrays;
```

Comment:
Will this become configurable?

Reply:
It's already configurable via the constructor below